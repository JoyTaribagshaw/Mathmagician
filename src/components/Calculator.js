import Button from './Button';
import Result from './Result';

function Calculator() {
  return (
    <div className="parent">
      <Result />
      <Button className="white-btn" value="AC" />
      <Button className="white-btn" value="+/-" />
      <Button className="white-btn" value="%" />
      <Button className="orange-btn" value="/" />
      <Button className="white-btn" value="7" />
      <Button className="white-btn" value="8" />
      <Button className="white-btn" value="9" />
      <Button className="orange-btn" value="x" />
      <Button className="white-btn" value="4" />
      <Button className="white-btn" value="5" />
      <Button className="white-btn" value="6" />
      <Button className="orange-btn" value="-" />
      <Button className="white-btn" value="1" />
      <Button className="white-btn" value="2" />
      <Button className="white-btn" value="3" />
      <Button className="orange-btn" value="+" />
      <Button className="white-btn zero" value="0" />
      <Button className="white-btn" value="." />
      <Button className="orange-btn" value="=" />
    </div>
  );
}

export default Calculator;
