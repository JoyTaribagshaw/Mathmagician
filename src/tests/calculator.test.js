import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  it('renders the Calculator component', () => {
    render(<Calculator />);
  });
});

it('renders correctly if there is no items present', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('checks UI changes state after the button is clicked', () => {
  const { getByText } = render(<Calculator />);
  const number1 = getByText(1);
  const answer = number1.textContent;
  expect(answer).toBe('1');
});
