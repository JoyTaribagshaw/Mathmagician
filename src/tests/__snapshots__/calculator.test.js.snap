// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="calculator_header"
  >
    <h1
      class="header"
    >
      Let us do some math!
    </h1>
    <div
      class="parent"
    >
      <div
        class="input"
      >
        0
      </div>
      <button
        class="white-btn"
        type="button"
      >
        AC
      </button>
      <button
        class="white-btn"
        type="button"
      >
        +/-
      </button>
      <button
        class="white-btn"
        type="button"
      >
        %
      </button>
      <button
        class="orange-btn"
        type="button"
      >
        /
      </button>
      <button
        class="white-btn"
        type="button"
      >
        7
      </button>
      <button
        class="white-btn"
        type="button"
      >
        8
      </button>
      <button
        class="white-btn"
        type="button"
      >
        9
      </button>
      <button
        class="orange-btn"
        type="button"
      >
        x
      </button>
      <button
        class="white-btn"
        type="button"
      >
        4
      </button>
      <button
        class="white-btn"
        type="button"
      >
        5
      </button>
      <button
        class="white-btn"
        type="button"
      >
        6
      </button>
      <button
        class="orange-btn"
        type="button"
      >
        -
      </button>
      <button
        class="white-btn"
        type="button"
      >
        1
      </button>
      <button
        class="white-btn"
        type="button"
      >
        2
      </button>
      <button
        class="white-btn"
        type="button"
      >
        3
      </button>
      <button
        class="orange-btn"
        type="button"
      >
        +
      </button>
      <button
        class="white-btn zero"
        type="button"
      >
        0
      </button>
      <button
        class="white-btn"
        type="button"
      >
        .
      </button>
      <button
        class="orange-btn"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly if there is no items present 1`] = `
<div
  className="calculator_header"
>
  <h1
    className="header"
  >
    Let us do some math!
  </h1>
  <div
    className="parent"
  >
    <div
      className="input"
    >
      0
    </div>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="white-btn zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="white-btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
